src/HealthChecks/HealthChecks/src/DependencyInjection/HealthCheckServiceCollectionExtensions.cs

```diff
@@ -26,7 +26,7 @@ public static class HealthCheckServiceCollectionExtensions
         public static IHealthChecksBuilder AddHealthChecks(this IServiceCollection services)
         {
             services.TryAddSingleton<HealthCheckService, DefaultHealthCheckService>();
-            services.TryAddSingleton<IHostedService, HealthCheckPublisherHostedService>();
+            services.TryAddEnumerable(ServiceDescriptor.Singleton<IHostedService, HealthCheckPublisherHostedService>());
```

Comment:
I wonder if AddHostedService should do this cc @Tratcher

Reply:
You think people are going to try adding the same hosted service multiple times? It hasn't come up yet.

Reply:
I guess the question is more about if people will use it to build libraries. If you wanted to use AddHostedService to add your hosted service and make your call idempotent, you have to revert to using this manual call.

Reply:
> I guess the question is more about if people will use it to build libraries.

Infinity times this. Library authors have to know about tryaddenumerable, or they have to throw when called twice.